code/modules/mob/living/carbon/life.dm

```diff
@@ -21,6 +21,7 @@
 
 	//Updates the number of stored chemicals for powers
 	handle_changeling()
+	handle_devil()

```

Comment:
That is exactly what's happening. It also constantly checks for changeling.
The soul counter only updates when a soul is being added or removed. I cannot think of a simple way to make it only CHECK when that happens, since I do not know how to reference procs from other files.